### PR TITLE
window.location DI snippet

### PR DESCRIPTION
**Pull request by fetis**
Shows how to inject window.location object

### CODE REVIEW

**Comment by thekiba** [comment]
Seems good! But I think we need something other example, because we have Location/PlatformLocation in Angular.

**Comment by fetis** [comment]
`Locaiton` is a part of standard `lib.dom.ts` and it differs from the Angular version. Maybe we should make a not about this, but I would keep it as, because it's a often use case.

File: snippets/window-dependency-injection.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ) {}
+
+  useIt() {
+    this.location.assign('xxx');
+  }
+}
+
+```
+
+It's also possible to provide this token on the app level by declaring it in your
```
Comment (NothingEverHappens) [inline snippets/window-dependency-injection.md]: Move to the bonus section and add an example?

File: snippets/window-dependency-injection.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+`AppModule`.
+
+# Links
+https://itnext.io/testing-browser-window-location-in-angular-application-e4e8388508ff
+https://angular.io/guide/dependency-injection
```
Comment (NothingEverHappens) [inline snippets/window-dependency-injection.md]: Please add an interactive example, see https://codelab-next.firebaseapp.com/angular/30-seconds/new

**Review by NothingEverHappens — CHANGES_REQUESTED** [review]

File: snippets/window-dependency-injection.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export const LOCATION_TOKEN = new InjectionToken<Location>('Window location object');
+
+@Component({
+  providers: [
+    { provide: LOCATION_TOKEN, useValue: window.location }
+  ]
+})
+export class SomeComponent {
+  constructor(
+    @Inject(LOCATION_TOKEN) private location: Location
```
Comment (NothingEverHappens) [inline snippets/window-dependency-injection.md]: Wait, why does the title say window, but instead location is injected?
We also need another snippet for window I guess?

Comment (fetis) [reply snippets/window-dependency-injection.md]: 🤦‍♂️ my fault. the idea is to provide `window.location`

**Comment by fetis** [comment]
@thekiba @NothingEverHappens 
ok, I cleaned it up and added code samples

**Review by kirjs — APPROVED** [review]

**Pull request merged**